wagtail/images/management/commands/wagtail_update_image_renditions.py

```diff
@@ -41,16 +41,14 @@ def handle(self, *args, **options):
         Rendition = get_image_model().get_rendition_model()
 
         renditions = Rendition.objects.all()
+        num_renditions = renditions.count()
 
         purge_only = options["purge_only"]
 
-        if not renditions.exists():
+        if num_renditions == 0:
             self.stdout.write(self.style.WARNING("No image renditions found."))
             return
 
-        rendition_ids = list(renditions.values_list("id", flat=True))
```

Comment:
I believe this is actually a feature - the ids need to be pre-calculated, otherwise the `.iterator` will see the newly-created renditions and never finish. See comment on L65. 

I think the solution here might be to run the loop in batches, rather than trying to load them all at once.